ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryLimitFunctionalTest.java

```diff
@@ -0,0 +1,491 @@
+                USER_PROVIDER,
+                FormatFactory.KAFKA,
+                FormatFactory.JSON,
+                timestampSupplier::getAndIncrement
+        );
+
+        //Create stream
+        makeAdminRequest(
+                REST_APP_0,
+                "CREATE STREAM " + USERS_STREAM
+                        + " (" + USER_PROVIDER.ksqlSchemaString(false) + ")"
+                        + " WITH ("
+                        + "   kafka_topic='" + topic + "', "
+                        + "   value_format='JSON');",
+                USER_CREDS
+        );
+        //Create table
+        output = KsqlIdentifierTestUtil.uniqueIdentifierName();
+        List<KsqlEntity> res = makeAdminRequestWithResponse(
+                REST_APP_0,
+                "CREATE TABLE " + output + " AS"
+                        + " SELECT " + USER_PROVIDER.key() + ", COUNT(1) AS COUNT FROM " + USERS_STREAM
+                        + " GROUP BY " + USER_PROVIDER.key() + ";",
+                USER_CREDS
+        );
+        queryId = extractQueryId(res.get(0).toString());
+        queryId = queryId.substring(0, queryId.length() - 1);
+        waitForTableRows();
+
+        waitForStreamsMetadataToInitialize(
+                REST_APP_0, ImmutableList.of(HOST0, HOST1, HOST2), USER_CREDS);
+    }
+
+    @After
+    public void cleanUp() {
+        REST_APP_0.closePersistentQueries(USER_CREDS);
+        REST_APP_0.dropSourcesExcept(USER_CREDS);
+        APP_SHUTOFFS_0.reset();
+        APP_SHUTOFFS_1.reset();
+        APP_SHUTOFFS_2.reset();
+    }
+
+    @Test
+    public void shouldReturnLimitRowsMultiHostSetupTable() {
+        // Given:
+        final int numLimitRows = 300;
+        final int limitSubsetSize = HEADER + numLimitRows;
+        ClusterFormation clusterFormation = findClusterFormation(TEST_APP_0, TEST_APP_1, TEST_APP_2);
+        waitForClusterToBeDiscovered(clusterFormation.router.getApp(), 3, USER_CREDS);
+        waitForRemoteServerToChangeStatus(clusterFormation.router.getApp(),
+                clusterFormation.router.getHost(), HighAvailabilityTestUtil.lagsReported(3), USER_CREDS);
+
+
+        final String sqlTableScan = "SELECT * FROM " + output + ";";
+        final String sqlLimit = "SELECT * FROM " + output + " LIMIT " + numLimitRows + " ;";
+
+        //issue table scan first
+        final List<StreamedRow> rows_0 = makePullQueryRequest(clusterFormation.router.getApp(),
+                sqlTableScan, null, USER_CREDS);
+
+        //check that we got back all the rows
+        assertThat(rows_0, hasSize(HEADER + TOTAL_RECORDS));
+
+        // issue pull query with limit
+        final List<StreamedRow> rows_1 = makePullQueryRequest(clusterFormation.router.getApp(),
+                sqlLimit, null, USER_CREDS);
+
+        // check that we only got limit number of rows
+        assertThat(rows_1, hasSize(limitSubsetSize));
+
+        // Partition off the active
+        clusterFormation.active.getShutoffs().shutOffAll();
```

Comment:
This is kinda cool to verify, but this is really an HA test of a sort.  Not sure if we want to include that in a limit test.